src/components/Card/Card.tsx

```diff
@@ -0,0 +1,148 @@
+import { useState, useEffect } from 'react';
+import './Card.css';
+import { useLocalStorage } from '../Utils/UseLocalStorage/UseLocalStorage.tsx';
+import { swapiFetch } from '../../services/swapiFetch/swapiFetch.ts';
+import { Outlet, useOutletContext, useParams } from 'react-router';
+import type { AppOutletContext } from '../App/App.tsx';
+import { useNavigate } from 'react-router';
+import { paginateArray } from '../../services/swapiFetch/paginateArray.ts';
+import { Pagination } from '../Pagination/Pagination.tsx';
+
+interface CardProps {
+  searchTerm: string;
+  toggleLoading: (isLoading: boolean) => void;
+}
+
+type ResultItem =
+  | { name: string; url: string; uid: string }
+  | { properties: { name: string; url: string }; uid: string };
+
+export default function Card() {
+  const [results, setResults] = useState<ResultItem[]>([]);
+  const [error, setError] = useState(false);
+  const [savedSearch] = useLocalStorage('searchTerm', '');
+  const { page, detailsId } = useParams<{
+    page?: string;
+    detailsId?: string;
+  }>();
+  const isDetailVisible = Boolean(detailsId);
+  const currentPage = page ?? '1';
+  const navigate = useNavigate();
+  const [loading, setLoading] = useState(true);
+  const { searchTerm, toggleLoading }: CardProps =
+    useOutletContext<AppOutletContext>();
+  const [isNextPage, setIsNextPage] = useState(false);
+
+  const handleClick = (uid: string) => {
+    navigate(`/${currentPage}/${uid}`);
+  };
+
+  const goToPage = (targetPage: number) => {
+    navigate(`/${targetPage}`);
+  };
+
+  const handlePrev = () => {
+    if (Number(currentPage) > 1) {
+      goToPage(Number(currentPage) - 1);
+    }
+  };
+
+  const handleNext = () => {
+    if (isNextPage) {
+      goToPage(Number(currentPage) + 1);
+    }
+  };
+
+  useEffect(() => {
+    setError(false);
+    toggleLoading(true);
+    setLoading(true);
+
+    const currentSearchTerm =
+      searchTerm !== undefined ? searchTerm : savedSearch;
+
+    swapiFetch(currentSearchTerm, currentPage)
+      .then((response) => {
+        toggleLoading(false);
+
+        let fetchedResults =
+          ('result' in response && response.result) ||
```

Comment:
consider using optional chaining 
```suggestion
          response?.result || response?.results || []
```